web/src/components/apps/Dashboard.jsx

```diff
@@ -26,65 +26,106 @@ const COMMON_ERRORS = {
   "no such host": "No such host",
 };
 
-class Dashboard extends Component {
-  state = {
-    appName: "",
-    iconUri: "",
-    currentVersion: {},
-    downstream: [],
-    links: [],
-    checkingForUpdates: false,
-    checkingUpdateMessage: "Checking for updates",
-    checkingForUpdateError: false,
-    appLicense: null,
-    activeChart: null,
-    crosshairValues: [],
-    noUpdatesAvalable: false,
-    updateChecker: new Repeater(),
-    uploadingAirgapFile: false,
-    airgapUploadError: null,
-    viewAirgapUploadError: false,
-    viewAirgapUpdateError: false,
-    airgapUpdateError: "",
-    startSnapshotErrorMsg: "",
-    showAutomaticUpdatesModal: false,
-    showAppStatusModal: false,
-    dashboard: {
-      appStatus: null,
-      metrics: [],
-      prometheusAddress: "",
-    },
-    getAppDashboardJob: new Repeater(),
-    fetchAppDownstreamJob: new Repeater(),
-    gettingAppLicenseErrMsg: "",
-    startSnapshotOptions: [
-      { option: "partial", name: "Start a Partial snapshot" },
-      { option: "full", name: "Start a Full snapshot" },
-      { option: "learn", name: "Learn about the difference" },
-    ],
-    selectedSnapshotOption: { option: "full", name: "Start a Full snapshot" },
-    snapshotDifferencesModal: false,
-  };
-
-  setWatchState = (app) => {
-    this.setState({
-      appName: app.name,
-      iconUri: app.iconUri,
-      currentVersion: app.downstream?.currentVersion,
-      downstream: app.downstream,
-      links: app.downstream?.links,
-    });
-  };
-
-  componentDidUpdate(lastProps) {
-    const { app } = this.props;
-    if (app !== lastProps.app && app) {
-      this.setWatchState(app);
-      this.getAppLicense(app);
+const updateCheckerRepeater = new Repeater();
+const getAppDashboardJobRepeater = new Repeater();
+const fetchAppDownloadstreamJobRepeater = new Repeater();
+
+const Dashboard = ({
+  app,
+  cluster: clusterProp,
+  refreshAppData,
+  updateCallback,
+  toggleIsBundleUploading,
+  history,
+  match,
+  makeCurrentVersion,
+  redeployVersion,
+  isHelmManaged,
+  ping,
+  snapshotInProgressApps,
+  isVeleroInstalled,
+  isBundleUploading,
+  onUploadNewVersion,
+}) => {
+  const [appName, setAppName] = useState("");
```

Comment:
In the future, if it's easier, when I've refactored class components that have a lot of state, I use `useReducer` like this:

```
const [state, setState] = useReducer((prevState, nextState) => ({
	...prevState,
	...nextState
}, {
	* your initial state here *
});
```

as it allows me to easily swap out `this.setState(...)` for `setState(...)` instead of having to manually declare all of these. What you have obviously works too, just wanted to share in case you were unaware.